.github/actions/setup-env/action.yml

```diff
@@ -41,11 +41,11 @@ runs:
         pip install -U pip
         pip install pipenv pdm
 
-    # Node.js
     - name: Setup pnpm
       if: inputs.setup_nodejs == 'true'
-      uses: pnpm/action-setup@v3
-      # Automatically reads `packageManager` field from root `package.json`.
+      shell: bash
+      # Corepack automatically reads `packageManager` field from root `package.json`.
```

Comment:
This kind of comment is sort of just documenting what Corepack _is_ isn't it? Do we need this kind of comment? I'd argue it wasn't necessary for `pnpm/action-setup` either.

What's the benefit of using corepack here instead of the action, by the way? Just consistency or is there something else? The action does _quite_ a lot of things, including basic clean up steps that are apparently necessary for GitHub actions (store prune at the end of the job, for example).

Is corepack by itself, without any additional support steps or jobs, sufficient to actually replace what `pnpm/action-setup` does? It's not only installing pnpm, after all: https://github.com/pnpm/action-setup/blob/master/src/index.ts#L8-L17

Reply:
The main reason is consistency and the removal of one extra dependency that needs to be updated.

I've seen many large projects use `corepack enable` instead of `pnpm/action-setup`. Do you think we really need all of the steps that the action does? Is store pruning important during the CI, if we are discarding the containers at the end anyway?

Reply:
I have no idea what the purpose of store pruning is at the end of the action, all I know is that `pnpm/action-setup` does it, and lots of other actions have similar clean up type steps. I have not read about why that is or what the advice is for this (I've never written of this nature). To clarify, store pruning is on the host, not inside the docker images, so relevant for when we run `pnpm install` in the next step after this one.

Based on a quick search, I can't find anything clear or definitive on why clean up steps are necessary, other than suggestions that while runners are intended to be ephemeral, clean up helps prevent any accidental leaks between runners. In our case, the pnpm store shouldn't have an issue in that regard, not that I can think of off the top of my head. We don't download and proprietary packages from private repositories, for example, so it isn't like there is IP that could leak between runners if ephemerality didn't work as expected. It also seems like it might be more relevant for self-hosted runners (perhaps those have different behaviour in this regard).